imagery-cesium/plugins/entities-visualizer/base-entities-visualizer.ts

```diff
@@ -25,45 +30,213 @@ export abstract class BaseEntitiesVisualizer extends BaseImageryVisualizer {
+			this.fixStyleValues(feature, styleSettings);
+
+			let firstStyle: any = {};
+			let secondaryStyle: any = {};
+			let textStyle: any = {};
+
+			if (styleSettings.shadow) {
+				secondaryStyle.stroke = new Stroke({
+					color: styleSettings.shadow.stroke,
+					width: styleSettings.shadow['stroke-width']
+				});
+			}
+
+			if (styleSettings.stroke) {
+				const color = this.colorWithAlpha(styleSettings.stroke, styleSettings['stroke-opacity']);
+				const dash = styleSettings['stroke-dasharray'];
+				const lineDash = dash > 0 ? [dash, 10] : undefined;
+				const width = styleSettings['stroke-width'];
+				const lineCap = dash > 0 ? 'square' : undefined;
+
+				firstStyle.stroke = new Stroke({ color, lineDash, width, lineCap, lineDashOffset: 5 });
+			}
+
+			if (styleSettings.fill) {
+				const color = this.colorWithAlpha(styleSettings.fill, styleSettings['fill-opacity']);
+				firstStyle.fill = new Fill({ color });
+			}
+
+			if (styleSettings.icon) {
+				firstStyle.image = new Icon(styleSettings.icon);
+			}
+
+			if (styleSettings.circle) {
+				const radius = styleSettings.circle;
+				firstStyle.image = new Circle({
+					radius,
+					fill: firstStyle.fill,
+					stroke: firstStyle.stroke
+				});
+			}
+
+			if (styleSettings.geometry) {
+				secondaryStyle.image = firstStyle.image;
+				secondaryStyle.geometry = styleSettings.geometry
+			}
+
+			if ((styleSettings.label && styleSettings.label.text) && !feature.getProperties().labelTranslateOn) {
+				const fill = new Fill({ color: styleSettings.label.fill });
+				const stroke = new Stroke({
+					color: styleSettings.label.stroke ? styleSettings.label.stroke : '#fff',
+					width: styleSettings.label.stroke ? 4 : 0
+				});
+				const { label } = styleSettings;
+
+				textStyle.text = new Text({
+					overflow: label.overflow,
+					font: `${ styleSettings.label.fontSize }px Calibri,sans-serif`,
+					offsetY: <any>styleSettings.label.offsetY,
+					text: <any>label.text,
+					fill,
+					stroke
+				});
+				textStyle.geometry = (feature) => {
+					const { label } = feature.getProperties();
+					if (label.geometry) {
+						const oldCoordinates = label.geometry.getCoordinates();
+						const newCoordinates = [this.offset[0] + oldCoordinates[0], this.offset[1] + oldCoordinates[1]];
+						return new Point(newCoordinates);
+					}
+					return new Point(this.getCenterOfFeature(feature).coordinates)
+				};
+
+				firstStyle.geometry = (feature) => feature.getGeometry();
+			}
+
+			if (styleSettings['marker-color'] || styleSettings['marker-size']) {
+				const color = styleSettings['marker-color'];
+				const radius = MarkerSizeDic[styleSettings['marker-size']];
+				firstStyle.image = new Circle({ fill: new Fill({ color }), stroke: null, radius });
+			}
+
+
+			return [firstStyle, textStyle, secondaryStyle].map(style => isStyle ? new Style(style) : style);
+		}
+	 */
+
 	clearEntities() {
 		this.billboardCollection.removeAll();
```

Comment:
delete the billboardCollection if its not in use and clear the custom data source

Reply:
merge latest code before uploading